### PR TITLE
DOCSTEAM 59 content refresh concepts

### PR DESCRIPTION
**Pull request by catherinepope**
# Description
- Made concluding links more scannable.
- Added opening definition to Data Persistence section.
- Added extra definitions.

# Reasons
https://circleci.atlassian.net/browse/DOCSTEAM-59

### CODE REVIEW

**Comment by devinmarieb** [comment]
@catherinepope This looks good - I think all the code samples for Server 2 need to stay on v2, while Cloud and Server 3 examples are upgraded to v2.1. Sorry if that was confusing language before. I'll let @rosieyohannan verify though. If you already verified the version for Server 2 code samples this can be ignored.

@kbatuigas - It's also possible these changes might have conflicts with the current PRs open for the Images PR updates since this page seems to still have all the old images in them. We might just want to keep an eye on this Concepts page for merge conflicts now on those open PRs.

File: jekyll/_cci2/concepts.md [hunk 1]
```diff
@@ -780,10 +782,85 @@ Workspaces | Duration of workflow | Attach the workspace in a downstream contain
...
+See [Using Contexts]({{ site.baseurl }}/2.0/contexts/) for more information.
+
 
 ## See also
 {: #see-also }
 {:.no_toc}
 
-Refer to the [Jobs and Steps]({{ site.baseurl }}/2.0/jobs-steps/) document for a summary of how to use the `jobs` and `steps` keys and options.¬
\ No newline at end of file
+[Jobs and Steps]({{ site.baseurl }}/2.0/jobs-steps/) document for a summary of how to use the `jobs` and `steps` keys and options.
```
Comment (rosieyohannan) [inline jekyll/_cci2/concepts.md:866]: This is a page I was thinking we might archive soon once we have the capability. I wonder where we want people to head after reading the concepts page - the configuration cookbook maybe, or an examples overview. I'll have a think/

**Review by rosieyohannan — COMMENTED** [review]
This looks great Catherine, thank you! - just the updates Devin mentioned to keep version: 2 config examples for server customers, and a comment about where to send people next.

Also - I think it would be good to add a section for concurrency, and what that means specifically within CircleCI - running multiple jobs concurrently using workflows. The word is often interchangeable with parallelism but not in CCI language.

**Comment by catherinepope** [comment]
Thank you, Rosie! Have corrected the version for Server 2 and put concurrency in a separate ticket.

**Review by rosieyohannan — APPROVED** [review]
Maybe we can make a new ticket for the alphabetisation, I won't mind :-) Looks great!

**Pull request merged**